contrib/node/README.md

```diff
@@ -1,53 +1,81 @@
-Obviously this is not desirable. Over time, Pants support for NPM should improve to work with a
-fully functional package.json with no duplication of information in the BUILD file.
+`node_module`: A node module target.
+`node_preinstalled_module`: A prebuilt node_module that will be downloaded during resolve step.
+`node_bundle`: A bundle of node modules.
+`node_remote_module`: Remote third party module. [Not recommended, instead specify dependencies in package.json with lock file]
+`node_test`: Run Javascript tests defined in package.json.
 
 # Functionality
 
-Utility is still limited right now.
+There is limited support for `lint` and `fmt` that leverage the use of `eslint`. `eslint` global rules will need to be configured by the repo owner
+and currently does not support target-level rule overrides.
 
-## Node distribution bootstrapping
+## Source level dependencies
 
-Pants will bootstrap itself with its own copy of Node and NPM and use them to run commands.
+Source-level dependencies is currently supported within the `yarn` package manager. To specify a target-level dependency on
+another node_module target, you must:
+
+1. Specify the local dependency using the `file:` specifier following npm rules within the `package.json` with relative path
+to the target dependency
+
+2. The dependent target must have a valid target definition and package.json with the same package name as specified in the `file:` dependency
+
+3. In the BUILD definition, the node_module target must also specify the fully qualified pants target address for each `file:` dependency.
+(In the future, with dep inference, this step may no longer be necessary).
+
+See the examples directory for real examples.
+
+## Bootstrapping
+
+Pants will bootstrap itself with its own copy of Node.js, NPM, and Yarn package manager and use them to run commands.
 It runs those commands with the bootstrapped Node's bin directory at the front of the PATH.
 
+Distribution versions can be modified in `pants.ini`.
+
+## Package Management
+
+There are two supported package management tools: NPM and Yarn. You can specify which package manager is used to `install`
+your dependencies in your BUILD target definition.
+
+NPM 5.0+ and Yarn both support locking dependencies to specific versions through the `package-lock.json` and `yarn.lock` files respectively.
+To ensure determinism and reproducibility within Pants, these files should be checked into your repository and included in the the target definition.
+
 ## Resolve
 
-Pants can install remote and local Node modules into the .pants.d working directory with
-"./pants resolve [target]".
+Pants can install Node modules into the .pants.d working directory with
 
-You can resolve node_remote_module and node_module targets. Pants effectively walks forward from
-the edges of the dependency tree topological sort for a target and does an "npm install"
+	`./pants resolve [target]`
```

Comment:
Thank you for cleaning this section up... but I wonder whether you shouldn't fully replace the suggestion to run `resolve` with a suggestion to run `node-install`? I can't think of a reason to manually run `resolve`: folks would probably want to run `test` instead?